x-pack/plugins/apm/server/routes/mobile/get_mobile_filters.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  termQuery,
+  kqlQuery,
+  rangeQuery,
+} from '@kbn/observability-plugin/server';
+import {
+  DEVICE_MODEL_IDENTIFIER,
+  HOST_OS_VERSION,
+  NETWORK_HOST_CONNECTION_TYPE,
+  SERVICE_NAME,
+  SERVICE_VERSION,
+  TRANSACTION_TYPE,
+} from '../../../common/elasticsearch_fieldnames';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
+import { getProcessorEventForTransactions } from '../../lib/helpers/transactions';
+
+type MobileFiltersTypes =
+  | 'device'
+  | 'appVersion'
+  | 'osVersion'
+  | 'netConnectionType';
+type MobileFilters = Array<{
+  key: MobileFiltersTypes;
+  options: string[];
+  label: string;
+}>;
+
+export async function getMobileFilters({
+  kuery,
+  apmEventClient,
+  serviceName,
+  environment,
+  start,
+  end,
+  searchAggregatedTransactions,
+  transactionType,
+}: {
+  kuery: string;
+  apmEventClient: APMEventClient;
+  serviceName: string;
+  environment: string;
+  start: number;
+  end: number;
+  searchAggregatedTransactions: boolean;
+  transactionType: string;
+}): Promise<MobileFilters> {
+  const response = await apmEventClient.search('get_mobile_filters', {
+    apm: {
+      events: [getProcessorEventForTransactions(searchAggregatedTransactions)],
+    },
+    body: {
+      track_total_hits: false,
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            ...termQuery(SERVICE_NAME, serviceName),
+            ...termQuery(TRANSACTION_TYPE, transactionType),
+            ...rangeQuery(start, end),
+            ...environmentQuery(environment),
+            ...kqlQuery(kuery),
+          ],
+        },
+      },
+      aggs: {
+        devices: {
+          terms: {
+            field: DEVICE_MODEL_IDENTIFIER,
+            size: 10,
+          },
+        },
+        osVersions: {
+          terms: {
+            field: HOST_OS_VERSION,
+            size: 10,
+          },
+        },
+        appVersions: {
+          terms: {
+            field: SERVICE_VERSION,
+            size: 10,
+          },
+        },
+        netConnectionTypes: {
+          terms: {
+            field: NETWORK_HOST_CONNECTION_TYPE,
+            size: 10,
+          },
+        },
+      },
+    },
+  });
+
+  return [
+    {
```

Comment:
I do get the point that this enables dynamic addition of filters but I would have an object like below and then define the labels in the UI. Up to you though
`
{
  device: response.aggregations?.devices?.buckets?.map(
          ({ key }) => key as string
        ) || [],
osVersion: ...
...
}
`

Reply:
I started with that, but it adds a lot of boilerplate on the UI side. Since it's all of the same type (EuSelect) I decided to use this approach.